examples/cute/tutorial/sgemm_1_sycl.cpp

```diff
@@ -283,16 +303,12 @@ gemm_nt(int m, int n, int k,
   auto dimBlock = syclcompat::dim3(size(tC));
   auto dimGrid  = syclcompat::dim3(size(ceil_div(M, bM)), size(ceil_div(N, bN)));
 
-  auto event = syclcompat::launch<
-      gemm_device<decltype(prob_shape), decltype(cta_tiler),
-                  TA, decltype(dA), decltype(sA), decltype(tA),
-                  TB, decltype(dB), decltype(sB), decltype(tB),
-                  TC, decltype(dC), decltype(sC), decltype(tC),
-                  Alpha, Beta>>(dimGrid, dimBlock, prob_shape, cta_tiler,
-                                A, dA, sA, tA,
-                                B, dB, sB, tB,
-                                C, dC, sC, tC,
-                                alpha, beta);
+  auto event = syclcompat::launch(dimGrid, dimBlock, [=]
```

Comment:
I guess you only want to record this event if you are doing fine-grained profiling?
It is possible you could see a notable performance improvement by only recording events when required: see https://github.com/intel/llvm/pull/18021/files#r2048537558

Reply:
Also if an analogue to `cudaEventRecord` in sycl would be useful to you then you could request this extension.